packages/verkle/src/node/leafNode.ts

```diff
@@ -133,16 +133,23 @@ export class LeafNode extends BaseVerkleNode<VerkleNodeType.Leaf> {
     }
   }
 
-  // Set the value at the provided index from the values array and update the node commitments
-  // TODO: Decide whether we need a separate "deleteValue" function since it has special handling
-  // since we never actually delete a node in a verkle trie but overwrite instead
-  setValue(index: number, value: Uint8Array): void {
+  /**
+   * Set the value at the provided index from the values array and update the node commitments
+   * @param index the index of the specific leaf value to be updated
+   * @param value the value to insert into the leaf value at `index`
+   */
+  setValue(index: number, value: Uint8Array | VerkleLeafNodeValue): void {
+    let val
+    // `val` is a bytes representation of `value` used to update the cCommitment
+    if (value instanceof Uint8Array) val = value
+    else
+      val = value === VerkleLeafNodeValue.Untouched ? new Uint8Array(32) : createDeletedLeafValue()
```

Comment:
Could use `createUntouchedLeafValue() ` instead of new Uint8Array(32)

Reply:
Done.  Anything else you want to add before we merge?

Reply:
All good lgtm